pkg/resource/deploy/source_eval.go

```diff
@@ -91,11 +92,17 @@ func (src *evalSource) Iterate(
 
 	tracingSpan := opentracing.SpanFromContext(ctx)
 
+	// Decrypt the configuration.
+	config, err := src.runinfo.Target.Config.Decrypt(src.runinfo.Target.Decrypter)
+	if err != nil {
+		return nil, result.FromError(errors.Wrap(err, "failed to decrypt config"))
+	}
+
 	// First, fire up a resource monitor that will watch for and record resource creation.
 	regChan := make(chan *registerResourceEvent)
 	regOutChan := make(chan *registerResourceOutputsEvent)
 	regReadChan := make(chan *readResourceEvent)
-	mon, err := newResourceMonitor(src, providers, regChan, regOutChan, regReadChan, tracingSpan)
+	mon, err := newResourceMonitor(src, providers, regChan, regOutChan, regReadChan, opts, config, tracingSpan)
```

Comment:
Does this pass the unencrypted configuration over an insecure gRPC channel? If so, does that change our existing security posture?

Reply:
It does pass the unencrypted configuration over gRPC, but it does not change our security posture: we already pass the unencrypted configuration to the language host in the environment.